packages/moonstone/Scrollable/Scrollbar.js

```diff
@@ -18,131 +18,121 @@ import componentCss from './Scrollbar.module.less';
-		 *
-		 * @type {Object}
-		 * @property {Number}    clientHeight    The client height of the list.
-		 * @property {Number}    clientWidth    The client width of the list.
-		 * @public
-		 */
-		clientSize: PropTypes.shape({
-			clientHeight: PropTypes.number.isRequired,
-			clientWidth: PropTypes.number.isRequired
-		}),
-
-		/**
-		 * Adds the corner between vertical and horizontal scrollbars.
-		 *
-		 * @type {Booelan}
-		 * @default false
-		 * @public
-		 */
-		corner: PropTypes.bool,
-
-		/**
-		 * `true` if rtl, `false` if ltr.
-		 * Normally, [Scrollable]{@link ui/Scrollable.Scrollable} should set this value.
-		 *
-		 * @type {Boolean}
-		 * @private
-		 */
-		rtl: PropTypes.bool,
-
-		/**
-		 * Registers the ScrollButtons component with an
-		 * {@link core/internal/ApiDecorator.ApiDecorator}.
-		 *
-		 * @type {Function}
-		 * @private
-		 */
-		setApiProvider: PropTypes.func,
-
-		/**
-		 * The scrollbar will be oriented vertically.
-		 *
-		 * @type {Boolean}
-		 * @default true
-		 * @public
-		 */
-		vertical: PropTypes.bool
-	}
-
-	static defaultProps = {
-		corner: false,
-		vertical: true
-	}
-
-	constructor (props) {
-		super(props);
-
-		if (props.setApiProvider) {
-			props.setApiProvider(this);
-		}
-
-		this.scrollbarRef = React.createRef();
-		this.scrollButtonsRef = React.createRef();
-	}
-
-	componentDidMount () {
-		const {getContainerRef, showThumb, startHidingThumb, update: uiUpdate} = this.scrollbarRef.current;
-
-		this.getContainerRef = getContainerRef;
-		this.showThumb = showThumb;
-		this.startHidingThumb = startHidingThumb;
-		this.uiUpdate = uiUpdate;
-
-		const {isOneOfScrollButtonsFocused, updateButtons, focusOnButton} = this.scrollButtonsRef.current;
-
-		this.isOneOfScrollButtonsFocused = isOneOfScrollButtonsFocused;
-		this.update = (bounds) => {
-			updateButtons(bounds);
-			this.uiUpdate(bounds);
+const ScrollbarBase = memo(forwardRef((props, ref) => {
+	// Refs
+	const scrollbarRef = useRef();
+	const scrollButtonsRef = useRef();
+	// render
+	const {cbAlertThumb, clientSize, corner, vertical, ...rest} = props;
+
+	useImperativeHandle(ref, () => {
+		const {getContainerRef, showThumb, startHidingThumb, update: uiUpdate} = scrollbarRef.current;
+		const {isOneOfScrollButtonsFocused, updateButtons, focusOnButton} = scrollButtonsRef.current;
+		return {
+			getContainerRef,
+			showThumb,
+			startHidingThumb,
+			uiUpdate,
+			isOneOfScrollButtonsFocused,
+			update: (bounds) => {
+				updateButtons(bounds);
+				uiUpdate(bounds);
+			},
+			focusOnButton
 		};
-		this.focusOnButton = focusOnButton;
-	}
-
-	render () {
-		const {cbAlertThumb, clientSize, corner, vertical, ...rest} = this.props;
+	}, [scrollbarRef, scrollButtonsRef]);
+
+	return (
+		<UiScrollbarBase
+			corner={corner}
+			clientSize={clientSize}
+			css={componentCss}
+			ref={scrollbarRef}
+			vertical={vertical}
+			childRenderer={({thumbRef}) => ( // eslint-disable-line react/jsx-no-bind
+				<ScrollButtons
```

Comment:
I think it would be better `() => { return ....;}` instead of `() => ()` to debug.
It is not related with this PR. But I hope we could do it this time.

Reply:
👌